tests/js/server/aql/aql-queries-geo.js

```diff
@@ -1358,7 +1358,36 @@ function geoLegacyComparison() {
                                  RETURN d._key`, {}, false);
       assertEqual(0, b.length);
     },
+  };
+}
+
+function geoThisAndThatSuite() {
+
+  let coll;
 
+  const cn = "UnitTestsAhuacatlGeoThisAndThat";
+
+  return {
+    setUp : function() {
+      db._drop(cn);
+      coll = db._create(cn);
+      coll.ensureIndex({type:"geo", fields:["geo"], geoJson: true});
+    },
+
+    tearDown : function() {
+      db._drop(cn);
+    },
+
+    testNestedFor : function() {
```

Comment:
I also suggest adding a additional tests that check that the geo index is used for varying collection sizes (small, medium, large), and also when using some filter conditions on the collection. This is because index cost formulas may take into account the number of to-be-processed documents.
Another question is also which index should be used if there are other candidate indexes, e.g.
```
FOR e IN ${cn}
  FOR d IN ${cn}
    FILTER GEO_INTERSECTS(d.geo, e.geo)
    FILTER d.value == @value 
    RETURN ...
```
when there is a persistent index present on `value`.

Reply:
Indeed, I have added more tests.

Reply:
Done.